File: src/main/org/nlogo/api/ExtensionException.java
package org.nlogo.api ;

/**
 * Wrapper class for exceptions thrown by NetLogo extensions.
 */
public strictfp class ExtensionException
	extends Exception
{
   /**
    * Creates a new ExtensionException
    * @param message	error message displayed to NetLogo user
    */
	public ExtensionException( String message )
	{
		super( message ) ;
	}

   /**
    * Creates a new ExtensionException
    * @param ex extension to be wrapped
    */
	public ExtensionException( Exception ex )
	{
		this( ex.getMessage() ) ;
		setStackTrace( ex.getStackTrace() ) ;
	}
}

File: netlogo-core/src/main/api/ExtensionException.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.api

/** Wrapper class for exceptions thrown by NetLogo extensions. */
class ExtensionException(message: String, cause: Exception) extends Exception(message) {
  if (cause != null)
    setStackTrace(cause.getStackTrace)
  def this(message: String) = this(message, null)
  def this(cause: Exception) = this(cause.getMessage, cause)
}
